src/server/shell/basic_idle_handler.cpp

```diff
@@ -215,6 +267,16 @@ void msh::BasicIdleHandler::set_display_off_timeout(std::optional<time::Duration
             display_config_controller, mir_power_mode_off, multiplexer);
         observers.push_back(power_setter);
         idle_hub->register_interest(power_setter, off_timeout);
+        if (off_timeout == time::Duration{} && previous_off_timeout)
```

Comment:
Maybe it is "just me", but I find that `off_timeout == time::Duration{0}` better expresses a comparison to zero. (I know comparing to "default constructed" has the same meaning to the compiler but it takes more thought to read)